server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByWithoutValueFilterDataSet.java

```diff
@@ -176,60 +190,46 @@ public void initGroupBy(QueryContext context, GroupByTimePlan groupByTimePlan)
-        for (int i = 0; i < aggregations.size(); i++) {
-          int resultIndex = indexes.get(i);
-          curAggregateResults[resultIndex] = aggregations.get(i);
+      while (!isEndCal()) {
+        // get pre-aggregate results of non-aligned series
+        for (Map.Entry<PartialPath, List<Integer>> entry : pathToAggrIndexesMap.entrySet()) {
+          MeasurementPath path = (MeasurementPath) entry.getKey();
+          List<Integer> indexes = entry.getValue();
+          GroupByExecutor groupByExecutor = pathExecutors.get(path);
+          List<AggregateResult> aggregations =
+              groupByExecutor.calcResult(curPreAggrStartTime, curPreAggrEndTime);
+          for (int i = 0; i < aggregations.size(); i++) {
+            int resultIndex = indexes.get(i);
+            slidingWindowAggrQueues[resultIndex].update(aggregations.get(i).clone());
+          }
         }
-      }
-      // get aggregate results of aligned series
-      for (Map.Entry<AlignedPath, List<List<Integer>>> entry :
-          alignedPathToAggrIndexesMap.entrySet()) {
-        AlignedPath path = entry.getKey();
-        List<List<Integer>> indexesList = entry.getValue();
-        AlignedGroupByExecutor groupByExecutor = alignedPathExecutors.get(path);
-        List<List<AggregateResult>> aggregationsList =
-            groupByExecutor.calcAlignedResult(curStartTime, curEndTime);
-        for (int i = 0; i < path.getMeasurementList().size(); i++) {
-          List<AggregateResult> aggregations = aggregationsList.get(i);
-          List<Integer> indexes = indexesList.get(i);
-          for (int j = 0; j < aggregations.size(); j++) {
-            int resultIndex = indexes.get(j);
-            curAggregateResults[resultIndex] = aggregations.get(j);
+        // get pre-aggregate results of aligned series
+        for (Map.Entry<AlignedPath, List<List<Integer>>> entry :
+            alignedPathToAggrIndexesMap.entrySet()) {
+          AlignedPath path = entry.getKey();
+          List<List<Integer>> indexesList = entry.getValue();
+          AlignedGroupByExecutor groupByExecutor = alignedPathExecutors.get(path);
+          List<List<AggregateResult>> aggregationsList =
+              groupByExecutor.calcAlignedResult(curPreAggrStartTime, curPreAggrEndTime);
+          for (int i = 0; i < path.getMeasurementList().size(); i++) {
+            List<AggregateResult> aggregations = aggregationsList.get(i);
+            List<Integer> indexes = indexesList.get(i);
+            for (int j = 0; j < aggregations.size(); j++) {
+              int resultIndex = indexes.get(j);
+              slidingWindowAggrQueues[resultIndex].update(aggregations.get(j).clone());
+            }
           }
         }
+        updatePreAggrInterval();
+      }
+      for (int i = 0; i < curAggregateResults.length; i++) {
+        curAggregateResults[i] = slidingWindowAggrQueues[i].getAggregateResult().clone();
```

Comment:
The `clone()` implemented in `AggregateResult` is just a shallow copy, is it satisfied your requriements?

Reply:
Because the member variables in the class are all basic data types, a shallow copy is enough.